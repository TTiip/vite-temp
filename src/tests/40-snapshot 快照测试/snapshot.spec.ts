import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import Hello from './hello'

describe('测试快照', () => {
  test('snapshot', () => {
    const wrapper = mount(Hello)
    const wrapperStr = wrapper.html()
    const obj = {
      name: 'xxx',
      age: 1,
    }
    expect(obj).toMatchSnapshot()
    expect(wrapperStr).toMatchSnapshot()
  })

  test('inline snapshot', () => {
    const obj = {
      name: 'xxx',
      age: 1,
    }
    expect(obj).toMatchInlineSnapshot(`
      {
        "age": 1,
        "name": "xxx",
      }
    `)
  })

  test('file snapshot', () => {
    const wrapper = mount(Hello)
    const wrapperStr = wrapper.html()
    expect(wrapperStr).toMatchFileSnapshot('./hello.html')
  })
})
