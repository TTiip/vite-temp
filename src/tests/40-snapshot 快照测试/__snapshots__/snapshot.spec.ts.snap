// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`测试快照 > snapshot 1`] = `
{
  "age": 1,
  "name": "xxx",
}
`;

exports[`测试快照 > snapshot 2`] = `"<div>Hello</div>"`;
